crates/compiler/can/src/constraint.rs

```diff
@@ -81,12 +79,6 @@ impl Constraints {
         let pattern_eq = Vec::new();
         let cycles = Vec::new();
 
-        types.extend([
-            Cell::new(Types::EMPTY_RECORD),
-            Cell::new(Types::EMPTY_TAG_UNION),
-            Cell::new(Types::STR),
-        ]);
-
```

Comment:
this trick can still be used (so empty record would have `EitherIndex::from_right(Variable::EMPTY_RECORD)`. is it not worth it in practice? (I'd think that it would be)

Reply:
I think it would be but let's address in a follow up